### PR TITLE
Add PyPy jobs to PR CI runs

### PR DESCRIPTION
**Pull request by webknjaz**
Ref https://github.com/jazzband/pip-tools/issues/1375#issuecomment-813212856

**Changelog-friendly one-liner**: Added PyPy jobs to PR builds

##### Contributor checklist

- [ ] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Review by ssbarnea — APPROVED** [review]

File: tests/test_cli_compile.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 from .constants import MINIMAL_WHEELS_PATH, PACKAGES_PATH
 
+is_pypy = "__pypy__" in sys.builtin_module_names
```
Comment (atugushev) [inline tests/test_cli_compile.py:14]: I'm not sure what's the canonical way to detect python implementation, maybe [platform.python_implementation() == "PyPy"](https://docs.python.org/3/library/platform.html#platform.python_implementation)?

Comment (webknjaz) [reply tests/test_cli_compile.py:14]: That may work too.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -87,3 +87,35 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - Ubuntu
+          - MacOS
+          # TODO: fix test_realistic_complex_sub_dependencies test on Windows
+          # - Windows
+        python-version:
+          - pypy3
```
Comment (webknjaz) [inline .github/workflows/ci.yml:103]:
```suggestion
          - pypy3.7
```

File: tests/test_cli_compile.py (hunk 1 above)
Comment (atugushev) [reply tests/test_cli_compile.py:14]: Okay, feel free to ignore the suggestion 👍🏻

**Review by atugushev — APPROVED** [review]
LGTM 👍

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -87,3 +87,35 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - Ubuntu
+          - MacOS
+          # TODO: fix test_realistic_complex_sub_dependencies test on Windows
+          # - Windows
+        python-version:
+          - pypy3.7
```
Comment (webknjaz) [inline .github/workflows/ci.yml:103]:
```suggestion
          - pypy-3.7
```

**Pull request merged**